external-crates/move/crates/move-core-types/src/vm_status.rs

```diff
@@ -308,6 +308,16 @@ pub enum StatusCode {
     // or took too long to run based on metering policies
     PROGRAM_TOO_COMPLEX = 1130,
 
+    ENUM_TAG_MISMATCH = 1131,
+    ENUM_SWITCH_BAD_OPERAND = 1132,
+    ENUM_TYPE_MISMATCH = 1133,
+    DUPLICATE_ENUM_SWITCH_TAG = 1134,
+    PARTIAL_ENUM_SWITCH = 1135,
+    VARIANT_FIELDS_ALREADY_BORROWED_MUTABLE = 1136,
+    ZERO_SIZED_ENUM = 1137,
+    RECURSIVE_ENUM_DEFINITION = 1138,
+    MAX_VARIANTS_REACHED = 1139,
```

Comment:
We check this in limits here: https://github.com/MystenLabs/sui/blob/tzakian/move-enums-0/external-crates/move/crates/move-bytecode-verifier/src/limits.rs#L197

Reply:
Should we be checking this during deserialization or check bounds? since its a bit of a implementation detail limit currently. It feels more analogous to the `u16::max` checks than a normal limit check

Reply:
Chatted out-of-band, but commenting here for other folks: We also check this in the deserializer